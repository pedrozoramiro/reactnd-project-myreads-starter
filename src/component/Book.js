import React from "react";
import PropTypes from "prop-types";
import * as Const from "../utils/Const";

const Book = ({ book, handleChangeShelf }) => {
  return (
    <div className="book">
      <div className="book-top">
        <div
          className="book-cover"
          style={{
            width: 128,
            height: 193,
            backgroundImage: `url(${book.imageLinks.smallThumbnail})`
          }}
        />
        <div className="book-shelf-changer">
          <select
            value={book.shelf || 'none'}
            onChange={e => handleChangeShelf(book, e.target.value)}
          >
            <option disabled>
              Move to...
            </option>
            {Const.SHELVES.map(function(shelve) {
              return (
                <option key={shelve.name} value={shelve.name}>
                  {shelve.title}
                </option>
              );
            })}
            <option value="none">None</option>
          </select>
        </div>
      </div>
      <div className="book-title">{book.title}</div>
      <div className="book-authors">{book.authors}</div>
    </div>
  );
};


Book.defaultProps = {
  book:{ shelf:'none'}
};

Book.propTypes = {
  book: PropTypes.object.isRequired,
  handleChangeShelf: PropTypes.func.isRequired
};

export default Book;
