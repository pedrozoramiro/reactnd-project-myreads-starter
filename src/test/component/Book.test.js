import React from "react";
import { shallow, mount } from "enzyme";

import Book from "../../component/Book";

const mock = {
  book: {
    title: "Book",
    shelf: "read",
    imageLinks: { smallThumbnail: "" },
    authors: ["Ramiro Pedrozo"]
  },
  shelves: [
    { name: "currentlyReading", title: "Currently Reading" },
    { name: "wantToRead", title: "Want to Read" },
    { name: "read", title: "Read" }
  ],
  handleChangeShelf: jest.fn()
};

it("renders without crashing", () => {
  const bookWrapper = (
    <Book
      book={mock.book}
      shelves={mock.shelves}
      handleChangeShelf={mock.handleChangeShelf}
    />
  );
  expect(shallow(bookWrapper)).toMatchSnapshot();
});

it("Should call the handleChangeShelf", () => {
  const bookWrapper = (
    <Book
      book={mock.book}
      shelves={mock.shelves}
      handleChangeShelf={mock.handleChangeShelf}
    />
  );

  mount(bookWrapper).find("select").simulate("change");
  expect(mock.handleChangeShelf).toHaveBeenCalledTimes(1);
});
