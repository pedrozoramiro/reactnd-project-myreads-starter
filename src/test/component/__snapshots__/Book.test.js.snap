// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="book"
>
  <div
    className="book-top"
  >
    <div
      className="book-cover"
      style={
        Object {
          "backgroundImage": "url()",
          "height": 193,
          "width": 128,
        }
      }
    />
    <div
      className="book-shelf-changer"
    >
      <select
        onChange={[Function]}
        value="read"
      >
        <option
          disabled={true}
          value="none"
        >
          Move to...
        </option>
        <option
          key="currentlyReading"
          value="currentlyReading"
        >
          Currently Reading
        </option>
        <option
          key="wantToRead"
          value="wantToRead"
        >
          Want to Read
        </option>
        <option
          key="read"
          value="read"
        >
          Read
        </option>
        <option
          value="none"
        >
          None
        </option>
      </select>
    </div>
  </div>
  <div
    className="book-title"
  >
    Book
  </div>
  <div
    className="book-authors"
  >
    Ramiro Pedrozo
  </div>
</div>
`;
